src/libpgexporter/configuration.c

```diff
@@ -151,6 +152,35 @@ pgexporter_init_configuration(void* shm)
    return 0;
 }
 
+/*
+ *
+ */
+int
+pgexporter_validate_config_file(char* path)
+{
+   if (path == NULL || !pgexporter_exists(path))
+   {
+      return 1;
+   }
+
+   if (!pgexporter_is_file(path))
+   {
+      return 2;
```

Comment:
Usually when you return values like this, you use enums or macros containing these values (in C it's usually the latter). This way, the user of `pgexporter_validate_config_file` doesn't have to look into the implementation to guess what each error value means.

Reply:
It's also a good practice to use err codes already baked into standard C (and often the OS) instead of custom error codes. Take a look at `man 3 errno` (either in your terminal or [here](https://man7.org/linux/man-pages/man3/errno.3.html)). They're also often returned as negative values (like `return -ENOENT;`).